source/agora/node/Validator.d

```diff
@@ -264,11 +264,25 @@ public class Validator : FullNode, API
     }
 
     /// GET /public_key
-    public override PublicKey getPublicKey () pure nothrow @safe
+    public override Identity getPublicKey (PublicKey key) nothrow @safe
     {
+        import agora.flash.OnionPacket : generateSharedSecret;
+        import libsodium.crypto_auth;
         endpoint_request_stats.increaseMetricBy!"agora_endpoint_calls_total"(
             1, "public_key", "http");
-        return this.config.validator.key_pair.address;
+
+        Identity id = Identity(this.config.validator.key_pair.address);
+        if (key == PublicKey.init)
+            return id;
```

Comment:
That should never happen, could it ?

Reply:
Well, there is nothing keeping someone from calling this endpoint with `PublicKey.init`.